analysis/combiners.py

```diff
@@ -644,6 +645,169 @@ def explain_computation(self):
+        abs_error_linf_expected = probability_to_keep * (
+            metrics.per_partition_error_min + metrics.per_partition_error_max)
+        abs_error_l0_variance = probability_to_keep * metrics.std_cross_partition_error**2
+        abs_error_variance = probability_to_keep * (
+            metrics.std_cross_partition_error**2 + metrics.std_noise**2)
+        loc_cpe_ne = metrics.expected_cross_partition_error
+        std_cpe_ne = math.sqrt(metrics.std_cross_partition_error**2 +
+                               metrics.std_noise**2)
+        abs_error_quantiles = []
+        if metrics.noise_kind == pipeline_dp.NoiseKind.GAUSSIAN:
+            error_distribution_quantiles = scipy.stats.norm.ppf(
+                q=self._error_quantiles, loc=loc_cpe_ne, scale=std_cpe_ne)
+        else:
+            error_distribution_quantiles = probability_computations.compute_sum_laplace_gaussian_quantiles(
+                laplace_b=metrics.std_noise / math.sqrt(2),
+                gaussian_sigma=metrics.std_cross_partition_error,
+                quantiles=self._error_quantiles,
+                num_samples=10**3)
+        for quantile in error_distribution_quantiles:
+            error_at_quantile = probability_to_keep * (
+                float(quantile) + metrics.per_partition_error_min +
+                metrics.per_partition_error_max)
+            abs_error_quantiles.append(error_at_quantile)
+
+        abs_error_expected_w_dropped_partitions = probability_to_keep * (
+            metrics.expected_cross_partition_error +
+            metrics.per_partition_error_min + metrics.per_partition_error_max
+        ) + (1 - probability_to_keep) * -metrics.sum
+
+        # Relative error metrics
+        if metrics.sum == 0:  # For empty public partitions & partitions with zero sum, to avoid division by 0
+            rel_error_l0_expected = 0
+            rel_error_linf_expected = 0
+            rel_error_l0_variance = 0
+            rel_error_variance = 0
+            rel_error_quantiles = [0] * len(self._error_quantiles)
+            rel_error_expected_w_dropped_partitions = 0
+        else:
+            # We take the absolute value of the denominator because it can be
+            # negative.
+            rel_error_l0_expected = abs_error_l0_expected / abs(metrics.sum)
+            rel_error_linf_expected = abs_error_linf_expected / abs(metrics.sum)
+            rel_error_l0_variance = abs_error_l0_variance / (metrics.sum**2)
+            rel_error_variance = abs_error_variance / (metrics.sum**2)
+            rel_error_quantiles = [
+                error / abs(metrics.sum) for error in abs_error_quantiles
+            ]
+            rel_error_expected_w_dropped_partitions = abs_error_expected_w_dropped_partitions / abs(
+                metrics.sum)
+
+        # Noise metrics
+        noise_variance = metrics.std_noise**2
+
+        return AggregateErrorMetricsAccumulator(
+            num_partitions=1,
```

Comment:
please add todo: to compute `num_partitions`

Reply:
WDYM exactly?

Reply:
Ah, sry, I thought that this is computing metrics and we don't compute num_partitions. I see, that this is creating an accumulator and of course it corresponds to 1 partition